src/lib/pickles/wrap_verifier.ml

```diff
@@ -626,8 +808,218 @@ struct
         absorb sponge PC (Boolean.true_, x_hat) ;
         let w_comm = messages.w_comm in
         Vector.iter ~f:absorb_g w_comm ;
+        let joint_combiner =
+          match messages.lookup with
+          | None ->
+              Types.Opt.None
+          | Maybe (_, _) ->
+              failwith "TODO"
+          | Some l -> (
+              let absorb_sorted_first_part () =
+                Vector.iter l.sorted ~f:(fun z ->
+                    let z = Array.map z ~f:(fun z -> (Boolean.true_, z)) in
+                    absorb sponge Without_degree_bound z )
+              in
+              let absorb_sorted_second_part () =
+                match l.sorted_5th_column with
+                | None ->
+                    ()
+                | Maybe (b, z) ->
+                    let z = Array.map z ~f:(fun z -> (b, z)) in
+                    absorb sponge Without_degree_bound z
+                | Some z ->
+                    let z = Array.map z ~f:(fun z -> (Boolean.true_, z)) in
+                    absorb sponge Without_degree_bound z
+              in
+              match[@warning "-4"]
```

Comment:
Hehe, saw you

Reply:
I can't add the TODO without it :man_shrugging: